src/Struct/InAppPurchaseCartPositionStruct.php

```diff
@@ -31,7 +31,9 @@ private function __construct(
     public static function fromArray(array $data): self
     {
         $inAppPurchaseCartPosition = (new self())->assign($data);
-        $inAppPurchaseCartPosition->setInAppFeatureIdentifier($data['feature']['identifier'] ?? '');
+        if ($inAppPurchaseCartPosition->getInAppFeatureIdentifier() === '') {
```

Comment:
Why is this necessary? It seems unexpected that `fromArray' does not allow you to override the identifier

Reply:
If the data received already has `in-app-feature-identifier` then it will overwrite that value with empty if `$data['feature']['identifier']` is not set.

This is a fallback for when `in-app-feature-identifier` doesn't exist in the array.

Reply:
doesn't it make sense then to check on the data instead of the object? `if ($data['feature']['identifier']) { ... }`?

Reply:
Well no, I only want to fall back it if the object doesn't have `$inAppFeatureIdentifier` set. If that is set, I don't want to overwrite it with `$data['feature']['identifier']`.